### DISCUSSION TITLE
GoPro Sleep Mode

### DISCUSSION THREAD

**Discussion by oliiiica** [gopro/OpenGoPro#266] [Q&A]
Hi, 

I would like to ask regarding the sleep mode in the GoPro API. Is the sleep mode the same as power down because when I tried to send the command that was stated in the API as sleep, the GoPro powered down and the BLE was disconnected. According to the API documentation (I have some snippet of the documentation below - Advertisements part), as I understood, if the camera is put into sleep mode, you can still see the advertisement but not in my case.

_**Advertisements**
The camera will send BLE advertisements while it is ON and for the first 8 hours after the camera is put to sleep. During this time, the camera is discoverable and can be connected to. If the camera is in sleep mode, connecting to it will cause the camera to wake and boot up._

It states that if the camera is in sleep mode, connecting to it will cause the camera to wake up and boot. However, just as I have said, when I sent the command for sleep:  _Sleep | Put camera to sleep | 01:05_. The camera powered down and disconnected. When I try to scan for the camera's advertisement to connect and wake it up by connecting, I can't find the camera's advertisement. 

Hope someone would be able to shed some light on the right way of implementing the sleep mode (not power down) and to be able to wake up from sleep mode and connect to the camera successfully. 

I am using Hero11 Black.

TIA!

**Comment by tcamise-gpsw** [comment]
Hello. I just tested this out and do not see the same behavior that you do. My test is the following with Hero11 Black, not connected to power:
1. Connect to camera BLE and Wifi
2. Wait 5 seconds
3. Send sleep command. At this point, I see the camera disconnect and continue advertising
4. Repeat

FWIW I've attached the sniffer capture here but you'll need the [Ellisys software](https://www.ellisys.com/products/btr1/download.php) to view it:
[reconnect.zip](https://github.com/gopro/OpenGoPro/files/10107617/reconnect.zip)

What are you using to scan / connect for the devices?

**Reply by oliiiica** [reply]
Hi @tcamise-gpsw,

I am using an nRF52840DK. So for my test here is my process:
1. Open the DK and scan for the advertisement of the GoPro.
2. Open the GoPro and connect by going to Connections –> Connect Device –> Quik App. The GoPro will advertise and the two devices will connect, pair, and bond. 
3. Send BLE command for taking photos. I take 10 photos. 
4. After taking the photos, send BLE command for sleep. 
5. After 5 mins, the DK will try to reconnect with the GoPro to take another 10 photos. 

For 5, what happens is that I can see the GoPro advertising and I can see it connects to the DK but suddenly it disconnects right after the connection. In this case the GoPro is unable to wakeup and take photos.

**Comment by tcamise-gpsw** [comment]
Ok so it is advertising after sleep but it immediately disconnects after you reestablish the connection.  I'm not convinced that this disconnect is coming from the GoPro but we need some more information to prove this.

Can you get some type of logging from the nRF52840DK to see what is causing the disconnect?

**Reply by oliiiica** [reply]
Hi @tcamise-gpsw,

What command do you send to wake up the GoPro from sleep? Or what are the needed requirements/process to wake up the GoPro from sleep mode?

Thanks!

**Reply by tcamise-gpsw** [reply]
When it is sleeping, there is no BLE connection and it is just advertising waiting for something to connect to it. So all you need to do to wake it up is to establish the BLE connection.
